src/tox/execute/request.py

```diff
@@ -38,6 +44,9 @@ def __init__(
         self.env = env  #: the environment variables to use
         self.stdin = stdin  #: the type of standard input interaction allowed
         self.run_id = run_id  #: an id to identify this run
+        if allow is not None and "*" in allow:
+            allow = None  # if we allow everything we can just disable the check
+        self.allow = allow
```

Comment:
So, if I get it right, `allow` is usually an empty list, or when you use e.g. make, it would be `["make"]`.

In which case would `allow` directly be `None`?

Reply:
An empty allow list would mean nothing is allowed. Which I'd imagine is always a bad value. So that should never happen.

An allow containing ``*`` means allow everything, for which we use ``allow=None``. 

When invoking process calls where we don't want to check allow list (e.g. for the PEP-517 packaging calls) the caller does not set allow, and such remains `None` meaning don't do allow validation.

When executing commands from within tox run we inject the environments path arguments into the allow list, plus whatever the user sets. See from the file below:

```
    def _allow_externals(self) -> List[str]:
        result: List[str] = [f"{i}{os.sep}*" for i in self._paths]
        result.extend(i.strip() for i in self.conf["allowlist_externals"])
        return result
```

This means that only within tox commands, commands_pre and commands_post the allow will be:

```
.tox\py39\bin\*  # set by the framework
ls               # set by the user
```

while in other scopes remains disabled. If the user passes ``*`` then we disable allow validation.